skrub/_interpolation_join.py

```diff
@@ -0,0 +1,328 @@
+
+from skrub._table_vectorizer import TableVectorizer
+
+
+class InterpolationJoin(base.BaseEstimator):
+    """Join with a table augmented by machine-learning predictions.
+
+    This is similar to a usual equi-join, but instead of looking for actual
+    rows in the right table that satisfy the join condition, we estimate what
+    those rows would contain if they existed in the table.
+
+    Suppose we want to join a table ``buildings(latitude, longitude, n_stories)``
+    with a table ``annual_avg_temp(latitude, longitude, avg_temp)``. Our annual
+    average temperature table may not contain data for the exact latitude and
+    longitude of our buildings. However, we can interpolate what we need from
+    the data points it does contain. Using ``annual_avg_temp``, we train a
+    model to predict the temperature, given the latitude and longitude. Then,
+    we use this model to estimate the values we want to add to our
+    ``buildings`` table. In a way we are joining ``buildings`` to a virtual
+    table, in which rows for any (latitude, longitude) location are inferred,
+    rather than retrieved, when requested. This is done with::
+
+        InterpolationJoin(annual_avg_temp, on=["latitude", "longitude"]).fit_transform(
+            buildings
+        )
+
+    Parameters
+    ----------
+    right_table : DataFrame
+        The table to be joined to the argument of ``transform``.
+        ``right_table`` is used to train a model that takes as inputs the
+        contents of the columns listed in ``right_on``, and predicts the contents
+        of the other columns. In the example above, we want our transformer to
+        add temperature data to the table it is operating on. Therefore,
+        ``right_table`` is the ``annual_avg_temp`` table.
+
+    left_on : list of str, or str
+        The columns in the left table used for joining. The left table is the
+        argument of ``transform``, to which we add information inferred using
+        ``right_table``. The column names listed in ``left_on`` will provide the
+        inputs (features) of the interpolators at prediction (joining) time. In
+        the example above, ``left_on`` is ``["latitude", "longitude"]``, which
+        refer to columns in the ``buildings`` table. When joining on a single
+        column, we can pass its name rather than a list: ``"latitude"`` is
+        equivalent to ``["latitude"]``.
+
+    right_on : list of str, or str
+        The columns in ``right_table`` used for joining. Their number and types
+        must match those of the `left_on` columns in the left table. These
+        columns provide the features for the estimators to be fitted. As for
+        ``left_on``, it is possible to pass a string when using a single column.
+
+    on : list of str, or str
+        Column names to use both `left_on` and `right_on`, when they are the
+        same. Provide either `on` (only) or both `left_on` and `right_on`.
+
+    suffix : str
```

Comment:
So we should accept suffixes for both the left and right dataframe, as in pandas? btw note `fuzzy_join` has `suffixes`, but `Joiner` doesn't have such a parameter. the AggJoiner PR atm uses `suffix` and applies suffixes to the right tables only. polars uses `suffix`, only for the right table, but only applies it if not applying it would result in a duplicate name (it doesn't check that applying it doesn't cause a duplicate name)

Reply:
from the skrub meeting: we keep `suffix` that is applied to the right table column names (see #751 )